### PR TITLE
DDF-0000 Make pressing enter in the login password field submit a sig…

### PR DESCRIPTION
**Pull request by rymach**
…n in request

Screen Shot 2019-03-14 at 6 36 40 PM

#### Who is reviewing it? 

@BernardIgiri 
@leo-sakh 
@gjvera 
@GabrielFabian 
@partycrab
@Lambeaux 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@bdeining

#### How should this be tested?
try it in firefox, chrome and edge
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx [hunk 1]
```diff
@@ -79,6 +79,11 @@ export default hot(module)(
                 showLabel={false}
                 showValidationIssues={false}
                 onChange={handlePasswordChange}
+                onKeyPress={event => {
+                  if (event.key === 'Enter') {
```
Comment (BernardIgiri) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx]: ✏️ The better way to do this is to wrap the sign in form in form tags, specify a submit button (`type="submit"` and for non-submit buttons `type="button"`), and create an on submit handler (`<form onSubmit={...}>`).

Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx]: I think @BernardIgiri 's suggestion would be the way to go here.

Comment (rymach) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx]: I agree this is the right way to it. I did try those changes, but it was getting clobbered by other components. IMO any more effort spent on "doing it right", in this case, probably isn't worth it.

**Comment by mojogitoverhere** [comment]
build now

**Comment by mojogitoverhere** [comment]
hero in progress

**Comment by gjvera** [comment]
### 🎉 Hero Successful 🎉 
Was able to login by pressing `Enter` in the right slideout 

LGTM

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx (hunk 1 above)
Comment (BernardIgiri) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx]: Did you call `event.preventDefault()` in your `onSubmit` handler?

Comment (rymach) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user/user.tsx]: no

**Comment by Lambeaux** [comment]
❓ What's the outcome of @BernardIgiri's feedback above? I'm curious why that's considered a best practice and what we gain by doing that? What are we not getting by ignoring it?

**Comment by Lambeaux** [comment]
❓ When testing, did we try holding the ENTER button down?

**Comment by rymach** [comment]
@Lambeaux 
1 Tried doing it that way, no luck getting it work quickly - I don't think it's worth spending more time on  
2 Should be fine

**Pull request merged**